server/src/main/java/org/elasticsearch/cluster/coordination/ClusterFormationFailureHelper.java

```diff
@@ -150,7 +150,7 @@ String getDescription() {
 
                 if (INITIAL_MASTER_NODE_COUNT_SETTING.get(Settings.EMPTY).equals(INITIAL_MASTER_NODE_COUNT_SETTING.get(settings))
                     && INITIAL_MASTER_NODES_SETTING.get(Settings.EMPTY).equals(INITIAL_MASTER_NODES_SETTING.get(settings))) {
-                    bootstrappingDescription = "cluster bootstrapping is disabled on this node";
+                    bootstrappingDescription = "[" + INITIAL_MASTER_NODES_SETTING.getKey() + "] is empty on this node";
```

Comment:
We check if it's set here:

https://github.com/elastic/elasticsearch/blob/6e6e63d01da8fab85d3a69656416c39bfd8c9139/server/src/main/java/org/elasticsearch/cluster/coordination/ClusterBootstrapService.java#L84-L85

If it's not set, and nor are any of the other settings listed here, then we auto-bootstrap after a few seconds which leads to a different log message that starts `master not discovered or elected yet` and doesn't mention bootstrapping.

However if bootstrapping hasn't occurred by the time we write this message then this can't have happened, so the problem is actually that it's empty:

https://github.com/elastic/elasticsearch/blob/6e6e63d01da8fab85d3a69656416c39bfd8c9139/server/src/main/java/org/elasticsearch/cluster/coordination/ClusterBootstrapService.java#L142

Reply:
My comment was just targeted at the if clause: 

```
INITIAL_MASTER_NODES_SETTING.get(Settings.EMPTY).equals(INITIAL_MASTER_NODES_SETTING.get(settings))```